system/lib/compiler-rt/wasm_setjmp.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2024 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ *
+ * a runtime implementation for
+ * https://github.com/llvm/llvm-project/pull/84137
+ * https://docs.google.com/document/d/1ZvTPT36K5jjiedF8MCXbEmYjULJjI723aOAks1IdLLg/edit
+ *
+ * a plan is to replace the __USING_WASM_SJLJ__ part of emscripten_setjmp.c
+ * with this.
```

Comment:
I don't think this comment is necessary. Also, `__wasm_sjlj_setjmp` and `__wasm_sjlj_test` can be used with Emscripten SjLj too. Only `__wasm_sjlj_longjmp`, which uses Wasm `throw`, is exclusive to Wasm SjLj.